engine/src/main/java/com/agiletec/aps/system/common/AbstractGenericCacheWrapper.java

```diff
@@ -63,17 +64,14 @@ protected void insertAndCleanCache(Cache cache, Map<String, O> objects) {
     protected void insertAndCleanCache(Cache cache, Map<String, O> objects, String codesCacheKey, String cacheKeyPrefix) {
         List<String> oldCodes = (List<String>) this.get(cache, codesCacheKey, List.class);
         List<String> codes = new ArrayList<>();
-        Iterator<String> iter = objects.keySet().iterator();
-        while (iter.hasNext()) {
-            String key = iter.next();
-            cache.put(cacheKeyPrefix + key, objects.get(key));
-            if (null != oldCodes) {
-                oldCodes.remove(key);
-            }
-            codes.add(key);
+        for (Map.Entry<String, O> entry: objects.entrySet()) {
+            cache.put(cacheKeyPrefix + entry.getKey(), entry.getValue());
+            codes.add(entry.getKey());
         }
         cache.put(codesCacheKey, codes);
-        this.releaseObjects(cache, oldCodes, cacheKeyPrefix);
+        List<String> keysToRelease = oldCodes == null ? null :
+                oldCodes.stream().filter(c -> !codes.contains(c)).collect(Collectors.toList());
```

Comment:
codes can they be more than 100 ? if yes, we should generate a map of code to reduce the computational complexity at the previous `for` step

Reply:
Used directly the map passed as parameter.